packages/assets-controllers/src/token-prices-service/codefi-v2.ts

```diff
@@ -370,25 +389,52 @@ export class CodefiTokenPricesServiceV2
         // to keep track of them and make sure we return the original versions.
         const lowercasedTokenAddress =
           tokenAddress.toLowerCase() as Lowercase<Hex>;
-        const lowercasedCurrency =
-          currency.toLowerCase() as Lowercase<SupportedCurrency>;
 
-        const price =
-          pricesByCurrencyByTokenAddress[lowercasedTokenAddress]?.[
-            lowercasedCurrency
-          ];
+        const tokenData =
```

Comment:
It appears that if `pricesByCurrencyByTokenAddress[lowercasedTokenAddress]` is `undefined`, this variable becomes `{}`. This means that any of the properties that you are destructuring below could also be `undefined`. 

At first glance, it appears that you're adding these properties to `tokenPrice` to add `tokenPrice` to the final object later. But then, there's a check below that prevent `tokenPrice` from being added to the final object (`tokenPrice.value !== undefined`). That's great! But it's a pretty roundabout way of skipping all of this code if `pricesByCurrencyByTokenAddress[lowercasedTokenAddress]` is `undefined`. So, instead of `|| {}`, what are your thoughts on doing an early return here? Something like:

``` typescript
        const tokenData =
          pricesByCurrencyByTokenAddress[lowercasedTokenAddress];

        if (tokenData === undefined) {
          return obj;
        }
```

And then you can say:

``` typescript
        const { price, ...rest } = tokenData;

        const tokenPrice: TokenPrice<Hex, SupportedCurrency> = {
          tokenAddress,
          value: price,
          currency,
          ...rest
        };

        return {
          ...obj,
          [tokenAddress]: tokenPrice
        };
```

This should also save you from having to mark these properties as optional in the `TokenPrice` type.

(Although, see note elsewhere for what to call `tokenPrice` here. Perhaps something like `tokenMarketData` would be more apt, and then this could simply become:)

``` typescript
        const marketData =
          pricesByCurrencyByTokenAddress[lowercasedTokenAddress];

        if (marketData === undefined) {
          return obj;
        }

        const token: Token<Hex, SupportedCurrency> = {
          address,
          currency,
          marketData,
        };
```